src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import styles from './Pagination.module.scss';
+import Text from 'components/Text';
+import ArrowDownIcon from 'components/icons/ArrowDownIcon';
+import classNames from "classnames";
+
+export interface PaginationProps {
+    className?: string;
+    currentPage: number;
+    pagesCount: number;
+    onItemClick: (pageNumber: number) => void;
+}
+
+const RANGE = 2;
+
+const Pagination: React.FC<PaginationProps> = ({ className, pagesCount, currentPage, onItemClick }) => {
+    const activeLeftArrow = currentPage !== 1;
+    const activeRightArrow = currentPage !== pagesCount;
+
+    const itemClickWrapper = (item: PaginationItem) => {
+        if (item.pageNumber && !item.active) {
+            onItemClick(item.pageNumber);
+        }
+    }
+
+    const onLeftArrowClick = () => {
+        if (activeLeftArrow) {
+            onItemClick(currentPage - 1);
+        }
+    }
+
+    const onRightArrowClick = () => {
+        if (activeRightArrow) {
+            onItemClick(currentPage + 1);
+        }
+    }
+
+    return (
+        <div className={classNames(styles.wrapper, className)}>
+            <ArrowDownIcon className={classNames(styles.arrow, styles.__left, activeLeftArrow && styles.__active)}
+                           onClick={onLeftArrowClick}
+            />
+            {
+                getPaginationItems(pagesCount, currentPage).map((item, index) => (
+                    <div className={classNames(styles.item, item.active  && styles.__active)}
+                         key={index}
+                         onClick={() => itemClickWrapper(item)}
+                    >
+                        <Text view={'p-18'} className={styles.text} >
+                            {
+                                item.displayedText
+                            }
+                        </Text>
+                    </div>
+                ))
+            }
+            <ArrowDownIcon className={classNames(styles.arrow, styles.__right, activeRightArrow && styles.__active)}
+                           onClick={onRightArrowClick}
+            />
+        </div>
+    )
+}
+
+interface PaginationItem {
+    active?: boolean;
+    displayedText: string;
+    pageNumber?: number;
+}
+
+const ELLIPSIS =  getPaginationItem(undefined, false, '...');
+
+function getPaginationItems(pagesCount: number, currentPage: number): PaginationItem[] {
+    const items: PaginationItem[] = [];
+    const leftBound = currentPage > RANGE * 2 ? Math.max(1, currentPage - RANGE) : 1;
```

Comment:
здесь видимо логика отсталась избыточная после экспериментов и можно упростить до: 
```suggestion
    const leftBound = currentPage > RANGE * 2 ? currentPage - RANGE : 1;
```